src/uproot/source/http.py

```diff
@@ -383,7 +391,14 @@ def handle_no_multipart(self, source, ranges, futures, results):
             results[chunk.start, chunk.stop] = chunk.raw_data
             futures[chunk.start, chunk.stop]._run(self)
 
-    def handle_multipart(self, source, futures, results, response, ranges):
+    def handle_multipart(
+        self,
+        source: uproot.source.chunk.Source,
+        futures,
+        results,
+        response,
+        ranges: list[(int, int)],
+    ):
```

Comment:
The types of `futures` and `results` can be determined from the way this method is called (only in one place). The `results` is (if I remember right) a length-1 list that gets updated in-place because Pythons 2 and 3 differed in their uses of `global` and `nonlocal`, so we got the same effect—a mutable reference—by wrapping the object to change in a list.

```suggestion
    def handle_multipart(
        self,
        source: uproot.source.chunk.Source,
        futures,
        results,
        response: http.client.HTTPResponse,
        ranges: list[(int, int)],
    ):
```